cpp/src/join/sort_merge_join.cu

```diff
@@ -537,14 +706,102 @@ sort_merge_join::inner_join(table_view const& left,
     preprocessed_right._null_processed_table_view,
     preprocessed_left._null_processed_table_view,
     [this, stream, mr](auto& obj) {
-      auto [preprocessed_right_indices, preprocessed_left_indices] = obj(stream, mr);
+      auto [preprocessed_right_indices, preprocessed_left_indices] = obj.inner(stream, mr);
       postprocess_indices(*preprocessed_right_indices, *preprocessed_left_indices, stream);
       stream.synchronize();
       return std::pair{std::move(preprocessed_left_indices), std::move(preprocessed_right_indices)};
     },
     stream);
 }
 
+std::pair<std::unique_ptr<rmm::device_uvector<size_type>>,
+          std::unique_ptr<rmm::device_uvector<size_type>>>
+sort_merge_join::left_join(table_view const& left,
+                           sorted is_left_sorted,
+                           rmm::cuda_stream_view stream,
+                           rmm::device_async_resource_ref mr)
+{
+  cudf::scoped_range range{"sort_merge_join::left_join"};
+  // Sanity checks
+  CUDF_EXPECTS(left.num_columns() != 0,
+               "Number of columns in left keys must be non-zero for a join",
+               std::invalid_argument);
+  CUDF_EXPECTS(left.num_columns() == preprocessed_right._null_processed_table_view.num_columns(),
+               "Number of columns must match for a join",
+               std::invalid_argument);
+
+  // Preprocessing the left table
+  preprocessed_left._table_view = left;
```

Comment:
*nit*: identifiers with a leading underscore are reserved names for the standard library. So these identifiers are all technically UB.